File: frontend/lib/model/Usuario.dart
import 'package:flutter/foundation.dart';

class Usuario{
   int id;
   String nome;
   String email;
   String senha;
   String tipoUsuario;

  Usuario({
    required this.id,
    required this.nome,
    required this.email,
    required this.senha,
    required this.tipoUsuario
  });

  factory Usuario.fromJson(Map<String, dynamic> json){
    return Usuario(
      id: json['id_usuario'] as int,
      nome: json['nome'] as String,
      email: json['email'] as String,
      senha: json['senha'] as String,
      tipoUsuario: json['tipoUsuario'] as String
    );
  }



}

File: backend/src/main/java/br/com/csi/backend/model/Usuario.java
package br.com.csi.backend.model;
import javax.persistence.*;

import lombok.Data;







@Table(name = "Usuario")
@Entity(name = "Usuario")
@Data
public class Usuario {
    
    @Id
    @GeneratedValue
    private Integer id_usuario;
    private String nome;
    private String email;
    private String senha;
    private String tipoUsuario;
    

}
